siva/repository.go

```diff
@@ -85,50 +75,48 @@ func (r *Repository) Mode() borges.Mode {
 
 // Commit implements borges.Repository interface.
 func (r *Repository) Commit() error {
-	if !r.transactional {
-		return borges.ErrNonTransactional.New()
-	}
-
-	if r.mode != borges.RWMode {
-		return nil
-	}
-
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
 	if r.closed {
 		return ErrRepoAlreadyClosed.New(r.id)
 	}
 
-	err := r.fs.Sync()
-	if err != nil {
-		return err
+	if !r.transactional {
+		return borges.ErrNonTransactional.New()
+	}
+
+	defer func() { r.closed = true }()
```

Comment:
Do we want the repository marked as closed if the `sto.Commit()` fails?

Reply:
Yes, I would mark it as closed as the transaction and maybe the siva itself would be in a consistent state and we don't want to let it do a second commit with broken data. I've also noticed that it needs to call `Rollback` if the commit fails.